packages/contracts-rfq/contracts/interfaces/IFastBridgeV2.sol

```diff
@@ -10,6 +10,37 @@ interface IFastBridgeV2 is IFastBridge {
         address relayer;
     }
 
+    /// @notice New params introduced in the FastBridgeV2.
+    /// We are passing fields from the older BridgeParams struct outside of this struct
+    /// for backwards compatibility.
+    /// Note: quoteRelayer and quoteExclusivitySeconds are either both zero (indicating no exclusivity)
+    /// or both non-zero (indicating exclusivity for the given period).
+    /// @param quoteRelayer             Relayer that provided the quote for the transaction
+    /// @param quoteExclusivitySeconds  Period of time the quote relayer is guaranteed exclusivity after user's deposit
+    /// @param quoteId                  Unique quote identifier used for tracking the quote
+    struct BridgeParamsV2 {
+        address quoteRelayer;
+        uint256 quoteExclusivitySeconds;
+        bytes quoteId;
+    }
+
+    /// @notice Updated bridge transaction struct to include parameters introduced in FastBridgeV2.
+    /// Note: only `exclusivityRelayer` can fill such a transaction until `exclusivityEndTime`.
+    /// TODO: consider changing the encoding scheme to prevent spending extra gas on decoding.
+    struct BridgeTransactionV2 {
+        BridgeTransaction txV1;
+        address exclusivityRelayer;
+        uint256 exclusivityEndTime;
+    }
+
+    event BridgeRequestedV2(bytes32 indexed transactionId, bytes quoteId);
```

Comment:
@parodime this would require all quote IDs to be exactly 32 bytes long, and right now I don't think this is finalised. Let's revisit that later on, given that's more of a gas optimisation question.

Reply:
Agree w @ChiTimesChi, think this makes sense but can revisit later

Reply:
Created SLT-286 to track this, resolving this thread here.